docs/filesystem.rst

```diff
@@ -0,0 +1,179 @@
+hierarchy, the file system is just a list of named files. Reading and writing
+a file is achieved via the standard Python ``open`` function and the resulting
+file-like object (representing the file) of types ``TextIO`` or ``BytesIO``.
+Operations for working with files on the file system (for example, listing or
+deleting files) are contained within `the os module <os.html>`__.
+
+If a file ends in the ``.py`` file extension then it can be imported.
+
+An example session in the MicroPython REPL may look something like this::
+
+    >>> with open('hello.py', 'w') as hello:
+    ...    hello.write("print('Hello')")
+    ...
+    >>> import hello
+    Hello
+    >>> with open('hello.py') as hello:
+    ...   print(hello.read())
+    ...
+    print('Hello')
+    >>> import os
+    >>> os.listdir()
+    ['hello.py']
+    >>> os.remove('hello.py')
+    >>> os.listdir()
+    []
+
+.. py:function:: open(filename, mode='r')
+
+    Returns a file object representing the file named in the argument
+    ``filename``. The mode defaults to ``'r'`` which means open for reading in
+    text mode. The other common mode is ``'w'`` for writing (overwriting the
+    content of the file if it already exists). Two other modes are available
+    to be used in conjunction with the ones describes above: ``'t'`` means
+    text mode (for reading and writing strings) and ``'b'`` means binary mode
+    (for reading and writing bytes). If these are not specified then ``'t'``
+    (text mode) is assumed. When in text mode the file object will be an
+    instance of ``TextIO``. When in binary mode the file object will be an
+    instance of ``BytesIO``.
+
+
+.. py:class:: TextIO
+
+    Instances of this class represent text files in the micro:bit's flat file
+    system.
+
+    You do not directly instantiate this class. Rather, an appropriately
+    configured instance of this class is returned by the ``open`` function
+    described above.
+
+    .. py:method:: close()
+
+        Flush and close the file. This method has no effect if the file is
+        already closed. Once the file is closed, any operation on the file
+        (e.g. reading or writing) will raise an exception.
+
+    .. py:method:: name()
+
+        Returns the name of the file the object represents. This will be the
+        same as the ``filename`` argument passed into the call to the ``open``
+        function that instantiated the object.
+
+    .. py:method:: read(size)
+
+        Read and return at most ``size`` characters from the file as a single
+        string. As a convenience, if ``size`` is unspecified or -1,
+        ``readall()`` is called. Fewer than ``size`` characters may be returned
+        if there are less than ``size`` characters remaining to be read from
+        the file.
+
+        If 0 characters are returned, and ``size`` was not 0, this indicates
+        end of file.
+
+    .. py:method:: readall()
```

Comment:
My god, you're right! OK... +1 on dropping the method (I guess in a different branch).